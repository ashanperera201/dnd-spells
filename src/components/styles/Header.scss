@import "../../styles/responsive";

.header-container {
    text-align: start;

    @include respond(phone) {
        text-align: center !important;
    }
}
