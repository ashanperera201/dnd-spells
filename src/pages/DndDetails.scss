@import "../styles/responsive";

.details-container {
    width: 100%;

    &__header {
        @include respond(phone) {
            text-align: center !important;
        }
    }
}

.spell{
    @include respond(phone) {
        text-align: center !important;
        margin-left: 0 !important;
    }
}

.desc{
    @include respond(phone) {
        margin-left: 0 !important;
    }
}