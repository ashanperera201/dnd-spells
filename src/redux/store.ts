import { createStore, combineReducers } from 'redux';
import favSpellsReducer from './fav-spells/fav-spell.reducer';

const reducers = combineReducers({
    spells: favSpellsReducer,
})

const store = createStore(
    reducers,
)

export default store;