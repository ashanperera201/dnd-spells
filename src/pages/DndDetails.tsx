import React, { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';
import axios from 'axios';
import Typography from '@mui/material/Typography';
import './DndDetails.scss'

const DndDetails = (props: any) => {

    let location = useLocation();

    const [spellDetail, setSpellDetail] = useState<any>();

    useEffect(() => {
        const state: any = location.state;
        fetchDetails(state);
    }, [location]);

    const fetchDetails = async (state: any) => {
        if (state) {
            axios.get(`http://www.dnd5eapi.co${state.url}`).then(serviceRes => {
                if (serviceRes && serviceRes.data) {
                    setSpellDetail(serviceRes.data);
                }
            })
        }
    }


    return (
        <div className="details-container">
            <Typography variant='h3' margin={3} className="details-container__header">
                Spell Detail
            </Typography>
            <Typography variant='h6' fontWeight={'500'} marginLeft={3} className={'spell'}>
                <b>Spell Name</b> : {spellDetail?.name}
            </Typography>
            <Typography variant='h6' fontWeight={'500'} marginLeft={3} textAlign={'justify'} className={'desc'}>
                <b>Spell Description</b> : {spellDetail?.desc[0]}
            </Typography>
            <Typography variant='h6' fontWeight={'500'} marginLeft={3} textAlign={'justify'} className={'desc'}>
                <b>Higher Level</b> : {spellDetail?.higher_level[0]}
            </Typography>
            <Typography variant='h6' fontWeight={'500'} marginLeft={3} textAlign={'justify'} className={'desc'}>
                <b>Material</b> : {spellDetail?.material}
            </Typography>
            <Typography variant='h6' fontWeight={'500'} marginLeft={3} textAlign={'justify'} className={'desc'}>
                <b>Range</b> : {spellDetail?.range}
            </Typography>
        </div>
    )
}

export default DndDetails;