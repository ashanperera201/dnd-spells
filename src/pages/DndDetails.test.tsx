import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import DndDetails from './DndDetails';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { BrowserRouter, } from 'react-router-dom'
import store from '../redux/store';
import { Provider as ReduxProvider } from "react-redux";
import routeData from 'react-router';

const theme = createTheme();

const location = {
    pathname: "/details",
    search: '',
    hash: '',
    state: {
        index: "acid-arrow",
        name: "Acid Arrow",
        url: "/api/spells/acid-arrow",
    }
}

jest.mock('react-router', () => ({
    ...jest.requireActual("react-router") as {},
    useLocation: jest.fn().mockImplementation(() => {
        return location;
    })
}));

const detailPageRender = (
    <React.StrictMode>
        <BrowserRouter>
            <ReduxProvider store={store}>
                <ThemeProvider theme={theme}>
                    <DndDetails />
                </ThemeProvider>
            </ReduxProvider>
        </BrowserRouter>
    </React.StrictMode>
)

describe('DND details page', () => {
    const useLocation = jest.spyOn(routeData, 'useLocation');

    beforeEach(() => {
        useLocation.mockReturnValue(location as any);
    });

    it('Should be renders dnd detail page correctly.', () => {
        const tree = renderer.create(
            detailPageRender
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should be renders header.', () => {
        const { getAllByText } = render(detailPageRender);
        expect(getAllByText("Spell Detail")).toBeTruthy();
    })
});

