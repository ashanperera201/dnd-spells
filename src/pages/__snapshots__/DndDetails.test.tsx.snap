// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DND details page Should be renders dnd detail page correctly. 1`] = `
<div
  className="details-container"
>
  <h3
    className="MuiTypography-root MuiTypography-h3 details-container__header css-mn6ykh-MuiTypography-root"
  >
    Spell Detail
  </h3>
  <h6
    className="MuiTypography-root MuiTypography-h6 spell css-1r5gz7r-MuiTypography-root"
  >
    <b>
      Spell Name
    </b>
     : 
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-h6 desc css-u6jdkr-MuiTypography-root"
  >
    <b>
      Spell Description
    </b>
     : 
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-h6 desc css-u6jdkr-MuiTypography-root"
  >
    <b>
      Higher Level
    </b>
     : 
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-h6 desc css-u6jdkr-MuiTypography-root"
  >
    <b>
      Material
    </b>
     : 
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-h6 desc css-u6jdkr-MuiTypography-root"
  >
    <b>
      Range
    </b>
     : 
  </h6>
</div>
`;
